File: java/20210129/src/Ex01.java

public class Ex01 {

	public static void main(String[] args) {
		// + - * / %
		
		int num1 = 10;
		int num2 = 20;
		
		System.out.println("num1 + num2 =" +num1+num2);
		System.out.println(num1+"+ num2 =" +num1+num2);
		System.out.println(num1 + " + " +num2 + " = " +num1+num2); // 이렇게 하면 1020이 됨.
		System.out.println(num1 + " + " +num2 + " = " +(num1+num2));//이렇게 해야 10+20 이 30으로 인식됨.
		
		
		System.out.println(num1 + " + " +num2 + " = " +(num1+num2));
		System.out.println(num1 + " - " +num2 + " = " +(num1-num2));
		System.out.println(num1 + " * " +num2 + " = " +(num1*num2));
		System.out.println(num1 + " / " +num2 + " = " +(num1/num2));
		System.out.println(num1 + " / " +num2 + " = " +((double)num1/(double)num2));
		System.out.println(num1 + " % " +num2 + " = " +(num1%num2));
	}
}

File: c_work21/20210127/ex01.c

/*
#include <stdio.h>

void doA(int a)
{
	a=20;
}
void main()
{
	int a = 10;
	int num1=0, num2=0;

	print("%d",a); //10
	doA(a);
	print("%d",a); //10
}
*/

/*
#include <stdio.h>

void doA(int *a) //주소값을 받기 위해서 포인터를 사용함
{
	*a=20;
}
void main()
{
	int a = 10;
	int num1=0, num2=0;

	print("%d",a); //10
	doA(&a); //a의 주소값을 넘겨주는 것. 주소값은 10을 가르킴
	print("%d",a); //20
}
*/

#include <stdio.h>

void doA(int *a) //주소값을 받기 위해서 포인터를 사용함
{
	*a=20;
    printf("doA a value a = %d\n", a);
    printf("doA a point value *a = %d\n", a);
     }
void main()
{
	int a = 10;
	int num1=0, num2=0;

    printf("main address is\n");
    printf("&a = %d\n", &a);
    printf("main address is\n");
    printf("a = %d\n",a);

	doA(&a); //a의 주소값을 넘겨주는 것. 주소값은 10을 가르킴
}